api/v1alpha1/condition_types.go

```diff
@@ -44,6 +44,12 @@ const (
 	// ComponentDescriptorNotFoundReason is used when the component descriptor cannot be found.
 	ComponentDescriptorNotFoundReason = "ComponentDescriptorNotFound"
 
+	// ComponentVersionNotFoundReason is used when the component version cannot be found.
+	ComponentVersionNotFoundReason = "ComponentVersionNotFound"
+
+	// ComponentVersionNotReadyReason is used when the component version is not ready.
+	ComponentVersionNotReadyReason = "ComponentVersionNotReady"
```

Comment:
Errr, sorry, I'm not sure I'm following what you mean. Could you please rewrite this question? :D

Reply:
I think the issue is a `ComponentVersionNotFound`   condition can be true or false but the reason should tell us why.

Condition = what
Reason = why

At least that's my understanding of @souleb 's point

Reply:
I still don't know what to do about this. :D Like, the CV is not ready. The reason isn't known for us why not. It would be if there would be an error. We could try and parse its status I guess and put that into the message?

Reply:
Maybe just don't use ComponentVersionNotReady as a reason? We usually do know why it's not ready (it's because the component version wasn't found or some part of the reconcile flow failed).

Reply:
> Condition = what
Reason = why

Yes. The reason can be unknown.

Reply:
`ComponentVersionNotReady`  is sensible to use in controllers that depend-on ComponentVersions.